### PR TITLE
feat: [CI-9926]: CI build and push to GAR step

### PR DESCRIPTION
**Pull request by shaurya-harness**
### Summary

 CI build and push to GAR step

#### Screenshots

Screenshot 2023-10-18 at 3 12 11 PM
Screenshot 2023-10-18 at 2 57 49 PM

#### PR Checklist

<details>
<summary>Please check if your PR fulfils the following requirements</summary>

- [ ] Tests for the changes have been added. Ideally, include a test that fails without this PR but passes with it.
- [ ] Docs have been [added/updated](https://harness.atlassian.net/jira/software/c/projects/DOC/boards/40).
</details>

<details>
<summary>Use the following comments to re-trigger PR Checks</summary>

- Jest: `retrigger jest`
- Prettier: `retrigger prettier`
- Type Check: `retrigger typecheck`
- ESLint: `retrigger eslint`
- Sonar: `retrigger sonar`
- Standards: `retrigger standards`
- Build: `retrigger build`
- Title Check: `retrigger titlecheck`
- Feature Name Check: `trigger featurenamecheck`
- Coverage: `retrigger coverage`
- Rebase: `trigger rebase`
- Cypress Rest: `retrigger cypress-rest`
- Cypress CD: `retrigger cypress-cd`
- Cypress Pipeline: `retrigger cypress-pipeline`
- Cypress CV: `retrigger cypress-cv`
- Fix Prettier: `fix prettier`
- MessageMetadata: `trigger messagecheck`
- GitLeaks: `trigger gitLeaks`
</details>

## PR check failures and solutions
https://harness.atlassian.net/wiki/spaces/BT/pages/21106884744/PR+Checks+-+Failures+and+Solutions

### CODE REVIEW

**Review by vardanbansal-harness — APPROVED** [review]

File: src/modules/75-ci/components/PipelineSteps/GARStep/GARStepBase.tsx [hunk 1]
```diff
@@ -0,0 +1,149 @@
...
+import { Formik, FormikForm, Accordion, Container } from '@harness/uicore'
+import type { FormikProps } from 'formik'
+import { get } from 'lodash-es'
+import { Connectors } from '@platform/connectors/constants'
+import type { StepFormikFowardRef } from '@pipeline/components/AbstractSteps/Step'
+import { setFormikRef } from '@pipeline/components/AbstractSteps/Step'
+import { usePipelineContext } from '@pipeline/components/PipelineStudio/PipelineContext/PipelineContext'
+import { useStrings } from 'framework/strings'
+
+import StepCommonFields /*,{ /*usePullOptions }*/ from '@ci/components/PipelineSteps/StepCommonFields/StepCommonFields'
```
Comment (d1wilko) [inline src/modules/75-ci/components/PipelineSteps/GARStep/GARStepBase.tsx:18]: maybe remove the comment? - I appreciate it probably comes from a copy paste

**Review by d1wilko — APPROVED** [review]
looks good - 1 suggestion - but I'll not let it block the PR :)

File: src/modules/75-ci/components/PipelineSteps/GARStep/GARStepBase.tsx (hunk 1 above)
Comment (shaurya-harness) [reply src/modules/75-ci/components/PipelineSteps/GARStep/GARStepBase.tsx:18]: yes

File: src/modules/75-ci/components/PipelineSteps/GARStep/__test__/GARStep.test.tsx [hunk 2]
```diff
@@ -0,0 +1,503 @@
...
+
+  describe('Edit View', () => {
+    test('should render properly', () => {
+      const { container } = render(
+        <TestStepWidget initialValues={{}} type={StepType.GAR} stepViewType={StepViewType.Edit} />
+      )
+
+      expect(container.querySelector('input[name="spec.optimize"]')).toBeDefined()
+
+      expect(container).toMatchSnapshot()
```
Comment (PratyushGar) [inline src/modules/75-ci/components/PipelineSteps/GARStep/__test__/GARStep.test.tsx:67]: Can w assert on elements instead of snapshot testing in this file

**Review by PratyushGar — APPROVED** [review]

File: src/modules/75-ci/components/PipelineSteps/GARStep/__test__/GARStep.test.tsx (hunk 2 above)
Comment (shaurya-harness) [reply src/modules/75-ci/components/PipelineSteps/GARStep/__test__/GARStep.test.tsx:67]: sure, will update in a seperate PR

**Pull request merged**